libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs

```diff
@@ -94,6 +99,27 @@ public static async Task RunAsync(this Dialog dialog, ITurnContext turnContext,
             }
         }
 
+        /// <summary>
+        /// Obtain the CultureInfo in DialogContext.
+        /// </summary>
+        /// <param name="dialogContext">The dialogContext to extract information.</param>
+        /// <returns>A string representing the current locale.</returns>
+        public static string GetLocale(this DialogContext dialogContext)
+        {
+            string locale = null;
+            object turnContent = null;
+            var turnExists = dialogContext?.Context?.TurnState?.TryGetValue(Turn, out turnContent);
+            if (turnExists == true && turnContent != null)
+            {
+                if ((turnContent as JObject)?.SelectToken(LocalePath) is JValue localeValue)
+                {
+                    locale = localeValue.ToObject<string>();
+                }
+            }
+
+            return locale;
```

Comment:
umm... how about just this:
{
   return dc.State.GetValue<string>("turn.locale");
}
 ?

Reply:
added Locale property in TurnContext.
And now the scenario is to set the locale as 
```C#
await CreateFlow(async (turnContext, cancellationToken) =>
            {
                turnContext.Locale = "de-DE";
                await dm.OnTurnAsync(turnContext, cancellationToken: cancellationToken).ConfigureAwait(false);
            })
```

Initially, I used setProperty action to set an locale
```
{
    "$kind": "Microsoft.SetProperty",
    "property": "turn.activity.locale",
    "value": "de-DE"
}

```

Reply:
@cosmicshuai @tomlm if we change from turn.activity.locale to turn.locale, where we init turn.locale, in adapter? both init turn.locale and init thread.currentCulture, make sense?